src/main/java/fi/aalto/cs/apluscourses/presentation/base/BaseTreeViewModel.java

```diff
@@ -32,21 +32,31 @@ public void setSelectedItem(@Nullable SelectableNodeViewModel<?> selectedItem) {
    */
   public BaseTreeViewModel(@NotNull T model,
                            @Nullable List<SelectableNodeViewModel<?>> children,
-                           @NotNull Options options) {
+                           @Nullable Options options) {
```

Comment:
Did you consider using some kind of "allGoesOptions" instead of making this nullable? That way some null checks below could perhaps have been avoided.